webapp/src/containers/createIncident/index.tsx

```diff
@@ -0,0 +1,314 @@
+        return {isLoading, isSuccess, isError, data: data as RecordData, error: (apiErr as FetchBaseQueryError)?.data as APIError | undefined};
+    };
+
+    const getResultPanelPrimaryBtnActionOrText = useCallback((action: boolean) => {
+        if (apiError) {
+            return action ? hideModal : 'Close';
+        }
+
+        return action ? handleOpenIncidentModal : 'Create another incident';
+    }, [apiError]);
+
+    const createIncident = () => {
+        if (!shortDescription) {
+            setValidationError(Constants.RequiredMsg);
+            return;
+        }
+
+        if (!channel && showChannelPanel) {
+            setShowChannelValidationError(true);
+            return;
+        }
+
+        // Set the first impact and urgency values by default.
+        const payload: IncidentPayload = {
+            short_description: shortDescription,
+            description,
+            caller_id: caller ?? '',
+            channel_id: channel ?? currentChannelId,
+        };
+
+        setIncidentPayload(payload);
+        makeApiRequest(Constants.pluginApiServiceConfigs.createIncident.apiServiceName, payload);
+    };
+
+    const handleError = (error: APIError) => {
+        if (error.id === Constants.ApiErrorIdNotConnected || error.id === Constants.ApiErrorIdRefreshTokenExpired) {
+            dispatch(setConnected(false));
+        }
+
+        setApiError(error);
+        setShowResultPanel(true);
+    };
+
+    useEffect(() => {
+        const {isLoading, isError, isSuccess, error, data} = getIncidentState();
+        setShowModalLoader(isLoading);
+        if (isError && error) {
+            handleError(error);
+        }
+
+        if (isSuccess) {
+            setApiError(null);
+            if (!showChannelPanel) {
+                setShowResultPanel(true);
+                return;
+            }
+
+            const subscriptionEvents = [
+                SubscriptionEvents.STATE,
+                SubscriptionEvents.PRIORITY,
+                SubscriptionEvents.COMMENTED,
+                SubscriptionEvents.ASSIGNMENT_GROUP,
+                SubscriptionEvents.ASSIGNED_TO,
+            ];
+
+            const payload: CreateSubscriptionPayload = {
+                server_url: SiteURL ?? '',
+                is_active: true,
+                user_id: Cookies.get(Constants.MMUSERID) ?? '',
+                type: SubscriptionType.RECORD,
+                record_type: RecordType.INCIDENT,
+                record_id: data.sys_id || '',
+                subscription_events: subscriptionEvents.join(','),
+                channel_id: channel ?? currentChannelId,
+            };
+
+            setSubscriptionPayload(payload);
+            makeApiRequest(Constants.pluginApiServiceConfigs.createSubscription.apiServiceName, payload);
+        }
+    }, [getIncidentState().isError, getIncidentState().isSuccess, getIncidentState().isLoading]);
```

Comment:
should data and error be here also?

Reply:
I don't think there is any need for this as they will update on the change in the above values